Zilon.Core/Zilon.Core/Persons/SurvivalStat.cs

```diff
@@ -20,7 +20,28 @@ public SurvivalStat(int startValue)
         /// <summary>
         /// Текущее значение.
         /// </summary>
-        public int Value { get; set; }
+        public int Value
+        {
+            get
+            {
+                return Value;
+            }
+            set
+            {
+                if (value >= Range.Max)
+                {
+                    Value = Range.Max;
+                }
+                else if (value <= Range.Min)
+                {
+                    Value = Range.Min;
+                }
+                else
+                {
+                    Value = value;
+                }
+            }
+        }
```

Comment:
This is where the setter goes to itself, which ultimately leads to recursion without an exit condition and a StackOverflowException.

You need to enter the private _value field and work with it inside the setter.

Reply:
Ah, yes, you're right. Created a private _value.